atom/renderer/api/atom_api_renderer_ipc.cc

```diff
@@ -55,49 +61,47 @@ class IPCRenderer : public mate::Wrappable<IPCRenderer> {
         .SetMethod("sendToHost", &IPCRenderer::SendToHost)
         .SetMethod("invoke", &IPCRenderer::Invoke);
   }
+
   static mate::Handle<IPCRenderer> Create(v8::Isolate* isolate) {
     return mate::CreateHandle(isolate, new IPCRenderer(isolate));
   }
 
-  void Send(mate::Arguments* args,
-            bool internal,
+  void Send(bool internal,
             const std::string& channel,
             const base::ListValue& arguments) {
-    electron_browser_ptr_->Message(internal, channel, arguments.Clone());
+    electron_browser_ptr_->get()->Message(internal, channel, arguments.Clone());
```

Comment:
Now that the interface ptr is created on the background runner, which is a different sequence to that of the main runner. Is it okay to call methods on this ptr in the main sequence without calling `PassInterface` ? well that would technically create a new pipe which would defeat the purpose. Just to make sure is this a safe operation ?

Reply:
The type of this pointer changed to `ThreadSafeElectronBrowserPtr`, whose _raison d'etre_ is to allow calling methods from multiple threads. This doesn't create any new pipes.

Reply:
Ah okay it uses `mojo/public/cpp/bindings/thread_safe_interface_ptr.h` , wasn't aware of this helper.

Reply:
I wasn't either until I asked in #mojo on the Chromium Slack :)